### ISSUE TITLE
bodycheck: false positive

### ISSUE THREAD

**Issue by carnott-snap** [golangci/golangci-lint#637]
1. Version of golangci-lint: `golangci-lint --version` (or git commit if you don't use binary distribution)
```
golangci-lint has version v1.17.1 built from (unknown, mod sum: "h1:lc8Hf9GPCjIr0hg3S/xhvFT1+Hydass8F1xchr8jkME=") on (unknown)
```
2. Config file: `cat .golangci.toml`
```toml
[linters]
disable-all = true
enable = ["bodyclose"]
```
3. Go environment: `go version && go env`
```
go version go1.12.7 linux/amd64
```
```
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/user/.cache/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/user/go"
GOPROXY=""
GORACE=""
GOROOT="/home/user/.local/share/umake/go/go-lang"
GOTMPDIR=""
GOTOOLDIR="/home/user/.local/share/umake/go/go-lang/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/tmp/test/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build952158134=/tmp/go-build -gno-record-gcc-switches"
```
4. Verbose output of running: `golangci-lint run -v`
```
   INFO [config_reader] Config search paths: [./ /tmp/test /tmp /] 
INFO [config_reader] Used config file .golangci.toml 
INFO [lintersdb] Active 1 linters: [bodyclose]    
INFO [loader] Go packages loading at mode load types and syntax took 1.07877735s 
INFO [loader] SSA repr building timing: packages building 16.798722ms, total 264.711926ms 
INFO [runner] worker.4 took 925ns                 
INFO [runner] worker.2 took 2.697µs               
INFO [runner] worker.1 took 1.879µs               
INFO [runner] worker.3 took 12.425656ms with stages: bodyclose: 12.415659ms 
INFO [runner] Workers idle times: #1: 12.19142ms, #2: 12.355192ms, #4: 12.397259ms 
INFO [runner] processing took 167.599µs with stages: identifier_marker: 59.675µs, source_code: 35.165µs, skip_dirs: 21.742µs, autogenerated_exclude: 13.224µs, cgo: 11.885µs, path_prettifier: 7.738µs, nolint: 7.642µs, max_same_issues: 2.386µs, uniq_by_line: 2.029µs, max_from_linter: 1.354µs, filename_unadjuster: 1.352µs, path_shortener: 1.182µs, max_per_file_from_linter: 1.022µs, exclude: 358ns, diff: 356ns, skip_files: 269ns, exclude-rules: 220ns 
main.go:9:23: bodyclose: response body must be closed (bodyclose)
        resp, err := http.Get("http://example.com/")
                             ^
INFO File cache stats: 1 entries of total size 220B 
INFO Memory: 18 samples, avg is 165.7MB, max is 337.7MB 
INFO Execution took 1.750390145s          
```
5. This issue was also reported upstream: timakin/bodyclose#19
```go
resp, err := http.Get("http://example.com/")
if err != nil {
        // handle error
}
f := resp.Body.Close
defer f()
_, err = ioutil.ReadAll(resp.Body)
```

**Comment by DRK3** [comment]
This is still an issue for me in v1.31.0.

**Comment by bombsimon** [comment]
It seems like it's not fixed in the upstream repository so not much we could do. I will keep this issue closed and as soon as the linked issue is resolved we can bump the version in `golangci-lint`.
